### PR TITLE
Implement non-conservative forces and stresses

### PR DESCRIPTION
**Pull request by frostedoyster**

### CODE REVIEW

File: src/ML-METATOMIC/pair_metatomic.cpp [hunk 1]
```diff
@@ -531,20 +580,25 @@ void PairMetatomic::compute(int eflag, int vflag) {
...
 
         auto forces = forces_tensor.accessor<double, 2>();
-        for (int i=0; i<atom->nlocal + atom->nghost; i++) {
+        for (int i=0; i<atom->nlocal; i++) {
             atom->f[i][0] += forces[i][0];
             atom->f[i][1] += forces[i][1];
             atom->f[i][2] += forces[i][2];
         }
+        if (!mta_data->non_conservative) {
+            for (int i=atom->nlocal; i<atom->nlocal + atom->nghost; i++) {
```
Comment (Luthaf) [inline src/ML-METATOMIC/pair_metatomic.cpp]: could this use the same `atoms_to_update` pattern as the kokkos version? It is clearer when reading the code IMO

File: src/KOKKOS/pair_metatomic_kokkos.cpp [hunk 2]
```diff
@@ -163,6 +163,15 @@ void PairMetatomicKokkos<DeviceType>::compute(int eflag, int vflag) {
     );
     mta_data->evaluation_options->set_selected_atoms(selected_atoms);
 
+    if (mta_data->non_conservative) {
+        // disable gradient tracking
+        system->positions().set_requires_grad(false);
+        system->cell().set_requires_grad(false);
+        for (auto nl_options: system->known_neighbor_lists()) {
+            system->get_neighbor_list(nl_options)->values().set_requires_grad(false);
+        }
```
Comment (Luthaf) [inline src/KOKKOS/pair_metatomic_kokkos.cpp]: instead of removing gradients here, could you change the require_grad for positions/cell when constructing the system?

File: src/ML-METATOMIC/pair_metatomic.cpp [hunk 3]
```diff
@@ -459,6 +484,15 @@ void PairMetatomic::compute(int eflag, int vflag) {
     );
     mta_data->evaluation_options->set_selected_atoms(selected_atoms);
 
+    if (mta_data->non_conservative) {
+        // disable gradient tracking
+        system->positions().set_requires_grad(false);
+        system->cell().set_requires_grad(false);
+        for (auto nl_options: system->known_neighbor_lists()) {
+            system->get_neighbor_list(nl_options)->values().set_requires_grad(false);
+        }
```
Comment (Luthaf) [inline src/ML-METATOMIC/pair_metatomic.cpp]: Same as kokkos, I'd prefer removing the gradient in the system adapter

**Review by Luthaf — COMMENTED** [review]
looks good overall! Could you also send a PR to https://github.com/metatensor/metatomic updating the documentation (https://docs.metatensor.org/metatomic/latest/engines/lammps.html#how-to-use-the-code)

**Review by Luthaf — APPROVED** [review]
